### PR TITLE
server: fix report_snapshot

### PR DESCRIPTION
**Pull request by huachaohuang**

### CODE REVIEW

File: src/server/server.rs [hunk 1]
```diff
@@ -728,4 +733,40 @@ mod tests {
...
+                                 store::new_snap_mgr("", None))
+            .unwrap();
+
+        for i in 0..10 {
+            let mut msg = Message::new();
+            if i % 2 == 1 {
+                msg.set_raft(RaftMessage::new());
+            }
+            server.report_unreachable(ConnData::new(0, msg));
+            let counter = RAFT_STORE_MSG_COUNTER.with_label_values(&["unreachable"]).get();
```
Comment (siddontang) [inline src/server/server.rs]: RAFT_STORE_MSG_COUNTER is a global value, other tests may also add it, so we can't use this for check directly.

Comment (siddontang) [inline src/server/server.rs]: I think we can test it with a mock RaftStoreRouter.

**Comment by siddontang** [comment]
CI failed @huachaohuang

**Comment by siddontang** [comment]
LGTM

**Comment by huachaohuang** [comment]
PTAL

**Pull request merged**